src/model/progression.rs

```diff
@@ -40,6 +47,18 @@ impl fmt::Display for Progression {
     }
 }
 
+impl<'de> Deserialize<'de> for Progression {
+    fn deserialize<TDeserializer>(
+        deserializer: TDeserializer,
+    ) -> std::result::Result<Self, TDeserializer::Error>
+    where
+        TDeserializer: Deserializer<'de>,
+    {
+        let progression_string: String = Deserialize::deserialize(deserializer)?;
+        Ok(Progression::parse(&progression_string).unwrap())
```

Comment:
This `.unwrap()` + `Ok()` is probably the wrong idea here - it should probably feed a parsing error into serde somehow (which didn't want to "just work" because the error type of the returned `Result` here isn't compatible with the error type returned by `Progression::parse()`)

I couldn't test this though because it seems like `Progression::parse()` doesn't actually fail when passed something that isn't a valid progression (opened #17)

Reply:
I think you can map your parsing error into a serde deserialization error:

```rust
use serde::de;

Progression::parse(&progression_string).map_err(de::Error::custom)?
```

Reply:
Ok updated to use `.map_err()`